src/common/interfaces/event.interface.ts

```diff
@@ -0,0 +1,223 @@
+  event: 'dsync.group.updated';
+  data: EventDirectoryGroupResponse & Record<'previous_attributes', any>;
+}
+
+export interface DsyncGroupUserAddedEvent extends EventBase {
+  event: 'dsync.group.user_added';
+  data: {
+    directoryId: string;
+    user: DirectoryUser;
+    group: Pick<EventDirectoryGroup, 'id' | 'name'>;
+  };
+}
+
+export interface DsyncGroupUserAddedEventResponse extends EventResponseBase {
+  event: 'dsync.group.user_added';
+  data: {
+    directory_id: string;
+    user: DirectoryUserResponse;
+    group: Pick<EventDirectoryGroupResponse, 'id' | 'name'>;
+  };
+}
+
+export interface DsyncGroupUserRemovedEvent extends EventBase {
+  event: 'dsync.group.user_removed';
+  data: {
+    directoryId: string;
+    user: DirectoryUser;
+    group: Pick<EventDirectoryGroup, 'id' | 'name'>;
+  };
+}
+
+export interface DsyncGroupUserRemovedEventResponse extends EventResponseBase {
+  event: 'dsync.group.user_removed';
+  data: {
+    directory_id: string;
+    user: DirectoryUserResponse;
+    group: Pick<EventDirectoryGroupResponse, 'id' | 'name'>;
+  };
+}
+
+export interface DsyncUserCreatedEvent extends EventBase {
+  event: 'dsync.user.created';
+  data: DirectoryUser;
+}
+
+export interface DsyncUserCreatedEventResponse extends EventResponseBase {
+  event: 'dsync.user.created';
+  data: DirectoryUserResponse;
+}
+
+export interface DsyncUserDeletedEvent extends EventBase {
+  event: 'dsync.user.deleted';
+  data: DirectoryUser;
+}
+
+export interface DsyncUserDeletedEventResponse extends EventResponseBase {
+  event: 'dsync.user.deleted';
+  data: DirectoryUserResponse;
+}
+
+export interface DsyncUserUpdatedEvent extends EventBase {
+  event: 'dsync.user.updated';
+  data: DirectoryUser & Record<'previousAttributes', any>;
+}
+
+export interface DsyncUserUpdatedEventResponse extends EventResponseBase {
+  event: 'dsync.user.updated';
+  data: DirectoryUserResponse & Record<'previous_attributes', any>;
+}
+
+export type Event =
+  | ConnectionActivatedEvent
+  | ConnectionDeactivatedEvent
+  | ConnectionDeletedEvent
+  | DsyncActivatedEvent
+  | DsyncDeactivatedEvent
+  | DsyncDeletedEvent
+  | DsyncGroupCreatedEvent
+  | DsyncGroupUpdatedEvent
+  | DsyncGroupDeletedEvent
+  | DsyncGroupUserAddedEvent
+  | DsyncGroupUserRemovedEvent
+  | DsyncUserCreatedEvent
+  | DsyncUserUpdatedEvent
+  | DsyncUserDeletedEvent;
+
+export type EventResponse =
+  | ConnectionActivatedEventResponse
+  | ConnectionDeactivatedEventResponse
+  | ConnectionDeletedEventResponse
+  | DsyncActivatedEventResponse
+  | DsyncDeactivatedEventResponse
+  | DsyncDeletedEventResponse
+  | DsyncGroupCreatedEventResponse
+  | DsyncGroupUpdatedEventResponse
+  | DsyncGroupDeletedEventResponse
+  | DsyncGroupUserAddedEventResponse
+  | DsyncGroupUserRemovedEventResponse
+  | DsyncUserCreatedEventResponse
+  | DsyncUserUpdatedEventResponse
+  | DsyncUserDeletedEventResponse;
+
+export type EventNames =
```

Comment:
Can we name this in the singular?

```suggestion
export type EventName =
```

Union types should use singular naming since they only can contain a single inhabitant value at a time.

Reply:
Sure can, copypasta :)